HACKING.md

```diff
@@ -0,0 +1,27 @@
+⚠⚠⚠ THIS IS A LIVING DOCUMENT AND LIKELY TO CHANGE QUICKLY ⚠⚠⚠
+
+# Hacking on the MCD
+
+1. Create a cluster (e.g. using [libvirt](https://github.com/openshift/installer/blob/master/Documentation/dev/libvirt-howto.md))
+2. Copy `/opt/tectonic/node-annotations.json` to `/etc/machine-config-daemon/` file on each node and pass it over this substitution:
+
+```
+sed -i s/node-configuration.v1.coreos.com/machineconfiguration.openshift.io/g /etc/machine-config-daemon/node-annotations.json
+```
+
+The install process will take care of this eventually.
+
+3. Create a namespace, e.g. `oc new-project mco`
+4. Create items from `manifests/` in a sensible order, e.g.
+
+```
+for f in manifests/*.crd.yaml; do oc create -f $f; done
+for f in manifests/*.machineconfigpool.yaml; do oc create -f $f; done
+oc create -f manifests/machineconfigdaemon/clusterrole.yaml
+oc create -f manifests/machineconfigdaemon/sa.yaml
+oc create -f manifests/machineconfigdaemon/scc.yaml
+oc create -f manifests/machineconfigdaemon/clusterrolebinding.yaml
+oc create -f manifests/machineconfigdaemon/daemonset.yaml
+```
+
+(Though note you'll have to sub in `{{.TargetNamespace}}` with `mco`).
```

Comment:
This looks to be still accurate even after https://github.com/openshift/machine-config-operator/pull/25/commits/e155824e24af3290d8923efc050c15a23a9012e4. The items at the manifest dir level no longer need a namespace, but items within subdirectories look like they still require it.

Reply:
@abhinavdahiya can you double check that?

Reply:
Yeah, that commit just removes it for the custom resources since the CRDs are now cluster scoped.

Reply:
WDYT if we copy valid mcd hacking manifests into `examples/**` directory. don't want people changing `manifests` directory.

Reply:
@abhinavdahiya as in providing a second set in the repo for people to modify? I worry that would cause maintenance burden when a change happened in in the `manifests`. However, if the instructions note to copy the `manifests` directory to `$LOCATION` before editing that would make sense to me.

Reply:
I left this out for now. It doesn't seem like a big deal either way? The diff will be tracked by `git status`, and plus not copying makes it easier to iterate on other fields in the manifests.